### PR TITLE
fix(generator): ensure warnings and errors are sent to std::clog

### PR DESCRIPTION
**Pull request by devbww**
If the generator is to use `GCP_LOG()` to report problems, then we must ensure that logging messages end up on the "stderr" output stream. The easiest way to do that is enable the `std::clog` backend on the `LogSink` (flush/tie aside), which we were already attempting.

Unfortunately the `EnableStdClog()` call was doing nothing because a default backend was already in place.  So, we must remove that backend first, and the easiest was to do that is to call `DisableStdClog()`.

Also lower the default minimum log severity to NOTICE, which is the first level that is an indication of a problem.

And finally add a `--log_level` flag so that the minimum can be lowered all the way down to the compile-time limit.  Parsing this flag required exposing `ParseSeverity()`.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11017)

### CODE REVIEW

File: google/cloud/log.h [hunk 1]
```diff
@@ -142,7 +143,10 @@ enum class Severity : int {
   GCP_LS_LOWEST_ENABLED = GOOGLE_CLOUD_CPP_LOGGING_MIN_SEVERITY_ENABLED,
 };
 
-/// Streaming operator, writes a human readable representation.
+/// Convert a human-readable representation to a Severity.
+absl::optional<Severity> ParseSeverity(std::string const& name);
```
[the remark this answers is not in the document]
Comment (devbww) [reply google/cloud/log.h:147]: Output streaming is already public, so the ~inverse probably should be too.  And customers would probably find it useful for the same reason we do.

File: generator/standalone_main.cc [hunk 2]
```diff
@@ -332,8 +334,21 @@ std::vector<std::future<google::cloud::Status>> GenerateCodeFromProtos(
 int main(int argc, char** argv) {
   int rc = 0;
   absl::ParseCommandLine(argc, argv);
-  google::cloud::LogSink::EnableStdClog(
-      google::cloud::Severity::GCP_LS_WARNING);
+  auto log_level = google::cloud::ParseSeverity(absl::GetFlag(FLAGS_log_level));
+  if (!log_level || *log_level > google::cloud::Severity::GCP_LS_NOTICE) {
+    log_level = google::cloud::Severity::GCP_LS_NOTICE;
+  }
+  google::cloud::LogSink::DisableStdClog();
```
[the remark this answers is not in the document]
Comment (devbww) [reply generator/standalone_main.cc:342]: Done.

File: generator/standalone_main.cc [hunk 3]
```diff
@@ -332,8 +334,21 @@ std::vector<std::future<google::cloud::Status>> GenerateCodeFromProtos(
...
+  google::cloud::LogSink::EnableStdClog(*log_level);
+  if (*log_level <
+      google::cloud::Severity::GOOGLE_CLOUD_CPP_LOGGING_MIN_SEVERITY_ENABLED) {
+    GCP_LOG(WARNING)
+        << "Log level " << *log_level
+        << " is less that the minimum enabled level of "
```
[the remark this answers is not in the document]
Comment (devbww) [reply generator/standalone_main.cc]: Done.

File: google/cloud/log.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (dbolduc) [reply google/cloud/log.h:147]: Ack. I don't see it tested anywhere in `log_test.cc`. Can you add a test for it?

[the remark this answers is not in the document]
Comment (devbww) [reply google/cloud/log.h:147]: Done.  Good catch.

**Review by dbolduc — APPROVED** [review]

**Pull request merged**